### ISSUE TITLE
data race detected

### ISSUE THREAD

**Issue by mberhault** [lightstep/lightstep-tracer-go#26]
We (cockroachdb) are running test builds with the race-detector enabled. These have caught a data race multiple times in lightstep.
Please see https://github.com/cockroachdb/cockroach/issues/8793 for details (and full trace log)

**Comment by tamird** [comment]
Looks like you guys have gone down the road of supporting GRPC to avoid the data races in thrift. According to @RaduBerinde using grpc results in the following log message:
```
LightStep instrumentation error (rpc error: code = 4 desc = context deadline exceeded). Set the Verbose option to enable more logging.
```
Is this something expected, or are we simply doing it wrong? The change was made in https://github.com/cockroachdb/cockroach/commit/0abeb5a48dcde340ca14bd09789857792b58455c.

cc @bensigelman @tschottdorf

**Comment by RaduBerinde** [comment]
Just to follow-up, the answer to the above question is here: https://github.com/cockroachdb/cockroach/pull/10121#issuecomment-255507645 and the issue was fixed in #54

**Comment by tedsuo** [comment]
Closing as this is resolved.
